src/reactor/segment_writer.rs

```diff
@@ -239,7 +263,7 @@ impl SegmentWriter {
                 SegmentWriter::MAX_WRITE_SIZE
             );
             if append.event.data.len() + to_send.len() <= SegmentWriter::MAX_WRITE_SIZE as usize
-                || event_count == SegmentWriter::MAX_EVENTS as usize
+                && event_count < SegmentWriter::MAX_EVENTS as usize
```

Comment:
While none of the current paths do this, we should handle that conditional and non-conditional appends are mixed, as this is logically possible given the API of this calss.

The logic here should be that conditional appends can be combined if their conditional lengths presume the prior append is a success. However conditional and non-conditional ones would need to be separate variables. 
So the condition should be added to this `if` so that is breaks out of the loop if it switches from conditional to non-conditional.

Reply:
On line 270 below, it could be:
`to_send.extend(&append.event.data);`
which would avoid making a copy.

Reply:
Thanks! I have updated the code to separate the conditional append with the regular append.